internal/pkg/cli/task_run.go

```diff
@@ -134,6 +175,172 @@ func (o *runTaskOpts) Ask() error {
 	return nil
 }
 
+// Execute create or update task resources and run the task.
+func (o *runTaskOpts) Execute() error {
+	if o.image == "" && o.dockerfilePath == "" {
+		o.dockerfilePath = defaultDockerfilePath
+	}
+
+	if o.imageTag == "" {
+		o.imageTag = defaultImageTag
+	}
+
+	if err := o.getNetworkConfig(); err != nil {
+		return err
+	}
+
+	cluster, err := o.getCluster()
+	if err != nil {
+		return err
+	}
+
+	if err := o.deployTaskResources(); err != nil {
+		return err
+	}
+
+	//if image is not provided, then we build the image and push to ECR repo
+	if o.image == "" {
+		uri, err := o.pushToECRRepo()
+		if err != nil {
+			return err
+		}
+		o.image = fmt.Sprintf(fmtImageURL, uri, o.imageTag)
+
+		// update image to stack
+		if err := o.updateTaskResources(); err != nil {
+			return err
+		}
+	}
+
+	return o.runTask(cluster)
+}
+
+func (o *runTaskOpts) getCluster() (string, error) {
+	if o.env == config.EnvNameNone {
+		cluster, err := o.ecsGetter.DefaultCluster()
+		if err != nil {
+			return "", fmt.Errorf("get default cluster: %w", err)
+		}
+		return cluster, nil
+	}
+
+	clusters, err := o.resourceGetter.GetResourcesByTags(clusterResourceType, map[string]string{
+		deploy.AppTagKey: o.AppName(),
+		deploy.EnvTagKey: o.env,
+	})
+
+	if err != nil {
+		return "", fmt.Errorf("get cluster by env %s: %w", o.env, err)
+	}
+
+	if len(clusters) == 0 {
+		return "", fmt.Errorf("no cluster found under env %s in app %s", o.env, o.AppName())
+	}
+	return clusters[0], nil
+}
+
+func (o *runTaskOpts) runTask(cluster string) error {
+	o.spinner.Start(fmt.Sprintf("Waiting for %s to be running.", o.groupName))
+	err := o.starter.RunTask(ecs.RunTaskInput{
+		Cluster:        cluster,
+		Count:          o.count,
+		Subnets:        o.subnets,
+		SecurityGroups: o.securityGroups,
+		TaskFamilyName: fmt.Sprintf(fmtTaskFamilyName, o.groupName),
+	})
+
+	if err != nil {
+		o.spinner.Stop(log.Serrorln(fmt.Sprintf("Failed to run %s", o.groupName)))
+		return fmt.Errorf("run task %s: %w", o.groupName, err)
+	}
+
+	o.spinner.Stop(log.Ssuccessln(fmt.Sprintf("Task(s) %s is running.", o.groupName)))
+	return nil
+}
+
+func (o *runTaskOpts) getNetworkConfig() error {
```

Comment:
should this be named `setNetworkConfig` since we don't return any values?

Reply:
right!!